gui/layers/_markers_layer.py

```diff
@@ -260,16 +269,37 @@ def _slice_markers(self, indices):
         # Get a list of the coords for the markers in this slice
         coords = self.coords
         if len(coords) > 0:
-            matches = np.equal(
-                coords[:, 2:],
-                np.broadcast_to(indices[2:], (len(coords), len(indices) - 2)))
-
-            matches = np.all(matches, axis=1)
-
-            in_slice_markers = coords[matches, :2]
-            return in_slice_markers, matches
+            if self._n_dimensional is False:
+                matches = np.equal(
+                    coords[:, 2:],
+                    np.broadcast_to(indices[2:], (len(coords), len(indices) - 2)))
+
+                matches = np.all(matches, axis=1)
+
+                in_slice_markers = coords[matches, :2]
+                return in_slice_markers, matches, 1
+            elif self.ndim > 2:
+                distances = abs(coords[:, 2:] -
+                                np.broadcast_to(indices[2:], (len(coords), len(indices) - 2)))
```

Comment:
Can't the broadcasting be implicit here?

Reply:
yup, fixed in a couple places, thanks